File: Example/Factory-Code/WaldhausenRealEstateFactory.cs
/// <summary>
/// @author Michael Hinterdorfer
/// </summary>
public class WaldhausenRealEstateFactory : RealEstateFactory
{

    public WaldhausenRealEstateFactory()
    {
    }

    private readonly int price_per_m2;
    private static readonly Location location = Location.Waldhausen;

    protected override RealEstate createRealEstate(RealEstateType type, double sqmeters, int rooms, double garden_sqmeters, int num_of_parkingslots)
    {
        double price = 0;
        price += sqmeters * price_per_m2;
        price += garden_sqmeters * price_per_m2;

        if (type == RealEstateType.Apartment)
        {
            Apartment apartment = new Apartment();

            apartment.setGarden_squaremeter(garden_sqmeters);
            apartment.setLocation(location);
            apartment.setNum_of_parkinglots(num_of_parkingslots);
            apartment.setRooms(rooms);
            apartment.setSquaremeter(sqmeters);
            apartment.setPrice(price);

            return apartment;
        }
        else if (type == RealEstateType.House)
        {
            House house = new House();

            house.setGarden_squaremeter(garden_sqmeters);
            house.setLocation(location);
            house.setNum_of_parkinglots(num_of_parkingslots);
            house.setRooms(rooms);
            house.setSquaremeter(sqmeters);
            house.setPrice(price);

            return house;
        }
        return null;
    }
}

File: Example/Factory-Code/WaldhausenRealEstateFactory.ts
/**
 * @author Michael Hinterdorfer
 */
class WaldhausenRealEstateFactory extends RealEstateFactory {
  /**
   * Default constructor
   */
  constructor() {
    super();
  }

  /**
   *
   */
  private price_per_m2: number = 6.5;
  private location: Location1 = Location1.Waldhausen;

  /**
   * @param type
   * @param sqmeters
   * @param rooms
   * @param garden_sqmeters
   * @param num_of_parkingslots
   * @return
   */
  createRealEstate(
    type: RealEstateType,
    sqmeters: number,
    rooms: number,
    garden_sqmeters: number,
    num_of_parkingslots: number
  ): RealEstate {
    let price: number = 0;
    price += sqmeters * this.price_per_m2;
    price += garden_sqmeters * this.price_per_m2;

    if (type == RealEstateType.Apartment) {
      let apartment: Apartment = new Apartment();
      apartment.garden_squaremeter = garden_sqmeters;
      apartment.location = this.location;
      apartment.num_of_parkinglots = num_of_parkingslots;
      apartment.rooms = rooms;
      apartment.squaremeter = sqmeters;
      apartment.price = price;
      return apartment;
    } else if (type == RealEstateType.House) {
      let house: House = new House();
      house.garden_squaremeter = garden_sqmeters;
      house.location = this.location;
      house.num_of_parkinglots = num_of_parkingslots;
      house.rooms = rooms;
      house.squaremeter = sqmeters;
      house.price = price;
      return house;
    }
    return null;
  }
}
